test/library.spec.js

```diff
@@ -24,12 +24,22 @@ describe('Parser', () => {
 
   describe('Variables', () => {
     it('strings', () => {
+      expect(parse('f')).to.be.equal('f');
       expect(parse('foo')).to.be.equal('foo');
     });
 
+    it('strings with spaces', () => {
+      expect(parse('"foo bar"')).to.be.equal('foo bar');
+      expect(parse('"foo bar     "')).to.be.equal('foo bar     ');
+      expect(parse('"  foo bar     "')).to.be.equal('  foo bar     ');
+    });
```

Comment:
Should we trim the leading and trailing spaces?

Reply:
I don't think so. I didn't because it's a perfectly valid way to name properties on an object:

![screenshot 2018-03-13 10 30 43](https://user-images.githubusercontent.com/404731/37358973-a638bd4a-26a9-11e8-8670-da7498a08c19.png)

Reply:
I think this is debatable. It's allowed in javascript, but we don't have to allow it. I think we should document what the rules are and enforce them. MathJS's acceptable characters are here: http://mathjs.org/docs/expressions/syntax.html#constants-and-variables

Of course we can't use these, for example `@` is not permitted by mathJS, but we must support it.

Reply:
> Contain only alpha characters (above) and digits

That's also a rule form mathJS that we can't follow, since we *should* support spaces in names ;)

I'm OK with saying we don't allow leading or trailing spaces, but there's no harm in allowing them either, as long as the value is quoted.

> We already restrict what character a variable can start with.

Not when it's quoted, at least not the way this PR is now. I can change it, just let me know your preference.